mkosi/__init__.py

```diff
@@ -3797,13 +3821,117 @@ def insert_verity(
             loopdev,
             args.verity_partno,
             verity,
-            root_partition_name(args, True),
+            root_partition_name(args, "Verity"),
             gpt_root_native(args.architecture, args.usr_only).verity,
             True,
             u,
         )
 
 
+def make_verity_sig(
+    args: CommandLineArguments, root_hash: Optional[str], do_run_build_script: bool, for_cache: bool
+) -> Tuple[Optional[BinaryIO], Optional[bytes], Optional[str]]:
+
+    if do_run_build_script or args.verity != "signed":
+        return None, None, None
+    if for_cache:
+        return None, None, None
+
+    assert root_hash is not None
+
+    with complete_step("Signing verity root hash…"):
+
+        PKCS7_NOCERTS  = 0x2    # Don't include signature certificates in result
```

Comment:
I need to double check, but this could be problematic when using cert chains (ie: the kernel keyring has a root cert, signing is done with a leaf)

Reply:
Kernel example is with this flag set, and the verification code as far as I read it in the kernel doesn't care.

Reply:
it will be fine for self-signed yeah, but does it still work if you sign with a leaf and have only the root in /etc/verity.d?

Reply:
no, but the code i wrote does not follow certificate chains right now. given that secureboot in uefi afair also disables all certificate chains logic i am not sure we should enable it. 

or to say this differently: let's start with the cert chains stuff off, we can still enable this later on any day.

Reply:
yeah that's fine, this is more of a reminder to myself to go and check what happens